src/api/model.ts

```diff
@@ -12,7 +12,10 @@ export type HTTPMETHOD =
 export type ApiError =
   | "METHOD NOT FOUND"
   | "PATH NOT FOUND"
-  | "STATUS NOT FOUND";
+  | "STATUS NOT FOUND"
+  | "REQUEST BODY NOT FOUND"
+  | "CONTENT WAS NOT FOUND"
+  | "BAD MIMETYPE";
```

Comment:
Reviewing this batch of PRs, I'm realizing that we probably should formulate these in a "neutral" fashion: "The spec says X but the request has Y" sort of messages. I think we should avoid assuming that either side is right.

Reply:
agreed, but I feel like we should rewrite the whole error handling once we have the basic features in place.

RIght now I'm not entirerly sure what I actually want reported back to me.

I'm thinking that we could borrow a few ideas from how compilers report errors maybe.